articles/what-is-jenkins/index.md

```diff
@@ -0,0 +1,61 @@
+Integration hell is a major problem to software projects and it is risky to their success reason. Seeing it was hard to estimate how long it would take to fix the things that went wrong. Developers had to figure out what went wrong, and that could be very hard to do in a large project. After diagnosing the problems, again time had to be spent fixing them, the project budget may also increase and the success of the project is risked. The solution to integration hell was to integrate continuously throughout the entire project, and thus the practice of Continuous Integration was introduced to the DevOps field.
+
+### What is Continuous Integration?
+CI is a development practice whereby developers are required to commit changes to the source code in a shared repository more frequently. CI says developers must integrate code early and often, so every time changes are pushed to the repository (such as GitHub) the project code should be checked into, compiled, and tested. The productivity of developers has been improved by Continuous Integration, rather than before when they used to wait until all components are developed. Then spend weeks and months working on bugs and errors that would [inevitably arise](https://www.edureka.co/blog/what-is-jenkins/).
+
+Continuous Integration allows the development teams to detect problems early on. It also automates the process of compiling, testing, and reporting the project source code. When a developer push changes into a repository, the CI server gets notified about the changes. The CI server either pulls the repository after a few minutes to see if there are any changes or the repository actively calls the CI server to let it know when there are changes. It then clones the repository and builds the project. If the project fails it notifies the team. The continuous integration server finally generates various reports such as the line coverage of the project test or style issues found on the [project code](https://www.edureka.co/blog/what-is-jenkins/).
+
+### Why Continuous Integration is needed
+- Using CI developers tend to experience fewer conflicts when integrating code, this is because when code get merged it does not easily break. Also, even if it breaks the problem is easily resolved.
```

Comment:
These sections too should come before issues.

Change the order of your paragraphs to ensure it flows.